File: Sieve_of_Eratosthenes.py
import math
def sieve(n):
    a = [0]*(n+1)
    for i in range(n+1):
        a[i] = i
    for i in range(2,int(math.sqrt(n))):
        if(a[i]>0):
            for j in range(i*i,n+1,i):
                a[j]=-1
                
    return a    
n = int(input("Enter your limit"))
mass=sieve(n)
print("All prime numbers up to your limit")
for i in range(2,n+1):
    if(mass[i]>0):
        print("", i," ",end="")

File: Sieve_of_Eratosthenes.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <cmath>
using namespace std;
void reweto(vector<int> & vec)
{
	for (int i = 2; i<sqrt(vec.size()); i++)
	{
		if (vec[i]>0)
		{
			for (int j = i*i; j <= vec.size(); j += i)
			{
				vec[j] = -1;
			}
		}

	}
}
int main()
{
	setlocale(0, "");
	vector<int> mass;
	int num = 0,item=0;

	cout << "Enter your limit"<<endl;
	cin >> num;

	for (int i = 0; i <= num; i++)
	{
		mass.push_back(i);
	}

	reweto(mass);
	
	cout << "All prime numbers up to your limit: ";

	for (int i = 2; i < mass.size(); i++)
		if (mass[i] > 0)
			cout << mass[i] << " ";
	system("pause");
	return 0;
}